understand-tidb/introduction-of-tidb-architecture.md

```diff
@@ -1,10 +1,157 @@
+| types/json             | JSON type related code                                                                                                                                                                                                                                                                         |
+| util                   | Some TiDB related tools. Many packages exist in this directory. Only some important packages are introduced in the following articles.                                                                                                                                                         |
+| util/admin             | Some methods used by the Admin statement of TiDB                                                                                                                                                                                                                                               |
+| util/charset           | Character set related code                                                                                                                                                                                                                                                                     |
+| util/chunk             | Chunk is a data expression structure introduced in TiDB 1.1. A Chunk stores several lines of data. During SQL computing, data flows among different modules in the unit of Chunk.                                                                                                              |
+| util/codec             | The encoding and decoding of various data types.                                                                                                                                                                                                                                               |
+
+At a glance, TiDB has 80 packages, which might let you feel overwhelmed, but not all of them are important, and some features only involve a small number of packages, so where to start to look at the source code depends on the purpose of looking at the source code.
+
+If you want to understand the implementation details of a specific feature, then you can refer to the module profiles above and just find the corresponding module.
+
+If you want to have a comprehensive understanding of the source code, then you can start from `tidb-server/main.go` and see how tidb-server starts and how it waits for and handles user requests. Then follow the code all the way through to see the exact execution of the SQL. There are also some important modules that need to be looked at to know how they are implemented. For the auxiliary modules, you can look at them selectively to get a general impression.
+
+### SQL Layer Architecture
+
+![sql-layer-architecture](https://user-images.githubusercontent.com/18818196/121646008-6e22a900-cac7-11eb-954a-fbca825855b0.png)
```

Comment:
It's a little out of date. E.g. the storage engine can be TiFlash, and there are more background workers now. But I think it's basically fine.

Reply:
@djshow832 ok. Would you mind to draw a picture (maybe by hands) or describe the architecture in the corresponding issue #23 or creating a new issue so that we can update it later?

Reply:
Creating an issue sounds fine to me. I'll do it.